index.go

```diff
@@ -633,61 +725,78 @@ border-radius: 3px;`
 		FirstChild: htmlg.Text(d.Text),
 	}
-	div := htmlg.DivClass("details",
-		code,
-	)
-	return []*html.Node{div}
+	return []*html.Node{code}
 }
 
-type commitsDetails struct {
+type commits struct {
 	Commits []*github.RepositoryCommit
 }
 
-func (d commitsDetails) Render() []*html.Node {
+func (d commits) Render() []*html.Node {
 	var nodes []*html.Node
 
 	for _, c := range d.Commits {
-		avatar := &html.Node{
-			Type: html.ElementNode, Data: atom.Img.String(),
-			Attr: []html.Attribute{
-				{Key: atom.Src.String(), Val: *c.Author.AvatarURL},
-				{Key: atom.Style.String(), Val: "width: 16px; height: 16px; vertical-align: top; margin-right: 6px;"},
-			},
-		}
-		sha := &html.Node{
-			Type: html.ElementNode, Data: atom.Code.String(),
-			FirstChild: htmlg.Text(shortSHA(*c.SHA)),
-		}
-		if c.HTMLURL != nil {
-			sha = &html.Node{
-				Type: html.ElementNode, Data: atom.A.String(),
-				Attr: []html.Attribute{
-					{Key: atom.Href.String(), Val: *c.HTMLURL},
-				},
-				FirstChild: sha,
-			}
-		}
-		message := &html.Node{
-			Type: html.ElementNode, Data: atom.Span.String(),
-			Attr: []html.Attribute{
-				{Key: atom.Style.String(), Val: "margin-left: 6px;"},
-			},
-			FirstChild: htmlg.Text(firstParagraph(*c.Commit.Message)),
-		}
-
-		div := htmlg.Div(avatar, sha, message)
+		div := htmlg.Div(commit{C: c}.Render()...)
 		div.Attr = append(div.Attr, html.Attribute{Key: atom.Style.String(), Val: "margin-top: 4px;"})
 		nodes = append(nodes, div)
 	}
 
-	div := htmlg.DivClass("details", nodes...)
-	return []*html.Node{div}
+	return nodes
+}
+
+type commit struct {
+	C     *github.RepositoryCommit
+	Short bool
+}
+
+func (c commit) Render() []*html.Node {
+	avatar := &html.Node{
+		Type: html.ElementNode, Data: atom.Img.String(),
+		Attr: []html.Attribute{
+			{Key: atom.Src.String(), Val: *c.C.Author.AvatarURL},
+			{Key: atom.Style.String(), Val: "width: 16px; height: 16px; vertical-align: top; margin-right: 4px;"},
+		},
+	}
+	sha := &html.Node{
+		Type: html.ElementNode, Data: atom.Code.String(),
+		FirstChild: htmlg.Text(shortSHA(*c.C.SHA)),
+	}
+	if c.C.HTMLURL != nil {
+		sha = &html.Node{
+			Type: html.ElementNode, Data: atom.A.String(),
+			Attr: []html.Attribute{
+				{Key: atom.Href.String(), Val: *c.C.HTMLURL},
+			},
+			FirstChild: sha,
+		}
+	}
+	message := &html.Node{
+		Type: html.ElementNode, Data: atom.Span.String(),
+		Attr: []html.Attribute{
+			{Key: atom.Style.String(), Val: "margin-left: 4px;"},
+			{Key: atom.Title.String(), Val: *c.C.Commit.Message},
+		},
+	}
+	switch c.Short {
+	case false:
+		message.AppendChild(htmlg.Text(firstParagraph(*c.C.Commit.Message)))
+	case true:
+		message.AppendChild(htmlg.Text(shortCommit(firstParagraph(*c.C.Commit.Message))))
+	}
+	return []*html.Node{avatar, sha, message}
 }
 
 func shortSHA(sha string) string {
 	return sha[:8]
 }
 
+func shortCommit(s string) string {
+	if len(s) <= 24 {
+		return s
+	}
+	return s[:23] + "…"
+}
```

Comment:
Perhaps `func ellipse(s string, len int) string` to replace `shortCommit` and `shortTitle`?